lab-doug/test/repair-router-test.js

```diff
@@ -0,0 +1,181 @@
+        repairCrud.createRepair({mechanicLastName: 'Remmy', repairName: 'strut replacement', laborCost: 75.00, partsCost: 375.00, receiptId: `${this.tempReceipt._id}`})
+        .then(repair => {
+          this.tempRepair = repair;
+          done();
+        });
+      }).catch(done);
+    });
+
+    after((done) => {
+      repairCrud.removeAllRepairs()
+      .then(() => done()).catch(done);
+    });
+
+    it('should return a 404 not found', (done) => {
+      request.get(`${baseUrl}/api/repair/${"123s456"}`)
+      .then(done)
+      .catch(err => {
+        const res = err.response;
+        expect(res.status).to.equal(404);
+        expect(res.text).to.equal('not found');
+        done();
+      });
+    });
+  });
+  describe('Testing PUT with valid id', function(){
+    before((done) => {
+      receiptCrud.createReceipt({customerLastName: 'Wilson', autoMake: 'VW', autoYear: 2015 })
+      .then(receipt => {
+        this.tempReceipt = receipt;
+      })
+      .then(() => {
+        repairCrud.createRepair({mechanicLastName: 'Remmy', repairName: 'strut replacement', laborCost: 75.00, partsCost: 375.00, receiptId: `${this.tempReceipt._id}`})
+        .then(repair => {
+          this.tempRepair = repair;
+          done();
+        });
+      }).catch(done);
+    });
+
+    it('should return a modified Repair object', (done) => {
+      request.put(`${baseUrl}/api/repair/${this.tempRepair._id}`)
+      .send ({mechanicLastName: 'TestName', repairName: 'test put repair', laborCost: 100.00, partsCost: 200.00, receiptId: `${this.tempReceipt._id}`})
+      .then((res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.repairName).to.equal('test put repair');
+        expect(res.body.laborCost).to.equal(100.00);
+        expect(res.body.partsCost).to.equal(200.00);
+        expect(res.body.receiptId).to.equal(`${this.tempReceipt._id}`);
+        done();
+      })
+      .catch(err => res.sendError(err));
+    });
+  });
+
+
+  describe('Testing DELETE with a valid id', function(req, res){
+    before((done) => {
+      receiptCrud.createReceipt({customerLastName: 'Wilson', autoMake: 'VW', autoYear: 2015 })
+      .then(receipt => {
+        this.tempReceipt = receipt;
+      })
+      .then(() => {
+        repairCrud.createRepair({mechanicLastName: 'Remmy', repairName: 'testing delete repair', laborCost: 75.25, partsCost: 375.25, receiptId: `${this.tempReceipt._id}`})
+        .then(repair => {
+          this.tempRepair = repair;
+          done();
+        });
+      }).catch(done);
+    });
+    it('should remove the document from the collection', (done) => {
+      request.del(`${baseUrl}/api/repair/${this.tempRepair._id}`)
+      .then((res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.repairName).to.equal('testing delete repair');
+        expect(res.body.laborCost).to.equal(75.25);
+        expect(res.body.partsCost).to.equal(375.25);
+        done();
+      }).catch(done);
```

Comment:
If you are getting lots of errors, and you don't know what they are, the way to handle it is to change your .catch(done) to a method that can console log the errors and then call done, otherwise you're just shooting in the dark.